### PR TITLE
Fix releasing CLIs pipeline

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Use the UPX 3.96
- Use the GoReleaser 1.1.0 
- Update docs: https://github.com/capactio/website/pull/89

## Testing
I tested that on my fork: https://github.com/mszostok/capact/runs/4558596650?check_suite_focus=true

The problem was that the on [`ubuntu-latest` the latest UPX has 3.95 version](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md)—in this version binaries compressed for macOS cannot be executed (https://github.com/upx/upx/issues/424)

## Related issue(s)

Fixes https://github.com/capactio/capact/issues/585

### CODE REVIEW

**Review by pkosiec — APPROVED** [review]
I confirm that it works now on my machine (Intel Macbook with macOS 12.0.1 Mojave), while the previous one didn't work (weird, probably it stopped working after my upgrade to Mojave). I built Capact CLI binary and compressed it with UPX 3.96 using our Goreleaser setup in Ubuntu Docker container.

What worries me is that multiple people [in the linked issue](https://github.com/upx/upx/issues/424) (and [another one](https://github.com/upx/upx/issues/434)) multiple people reported also problems with 3.96, probably in different setups (e.g. ARM-based Macs, different OS version etc.). Some people recommend using upx straight from `devel`.

Anyway, it definitely improves the situation on Intel-based Macs. We can merge this and bump UPX version later if a new version is released.

Please take a look on one minor comment before merge 🙂

File: .github/workflows/branch-build.yaml
Comment (mszostok) [reply .github/workflows/branch-build.yaml:194]: yes, sorry for that 👍

**Pull request merged**
